metric/doc.go

```diff
@@ -30,6 +30,64 @@ can then be built from the [Meter]'s instruments. See
 [go.opentelemetry.io/otel/metric/instrument] for documentation on each
 instrument and its intended use.
 
+# API Implementations
+
+This package does not conform to the standard Go versioning policy, all of its
+interfaces may have methods added to them without a package major version bump.
+This non-standard API evolution could surprise an uninformed implementation
+author. They could unknowingly build their implementation in a way that would
+result in a runtime panic for their users that update to the new API.
+
+The API is designed to help inform an instrumentation author about this
+non-standard API evolution. It requires them to choose a default behavior for
+unimplemented interface methods. There are three behavior choices they can
+make:
+
+  - Compilation failure
+  - Panic
+  - Default to another implementation
+
+All interfaces in this API embed a corresponding interface from
+[go.opentelemetry.io/otel/metric/embedded]. If an author wants the default
+behavior of their implementations to be a compilation failure, signaling to
+their users they need to update to the latest version of that implementation,
+they need to embed the corresponding interface from
+[go.opentelemetry.io/otel/metric/embedded] in their implementation. For
+example,
+
+	import "go.opentelemetry.io/otel/metric/embedded"
+
+	type MeterProvider struct {
+		embedded.MeterProvider
+		// ...
+	}
+
+If an author wants the default behavior of their implementations to a panic,
+they need to embed the API interface directly.
+
+	import "go.opentelemetry.io/otel/metric"
+
+	type MeterProvider struct {
+		metric.MeterProvider
+		// ...
+	}
+
+This is not a recommended behavior as it could lead to publishing packages that
+contain runtime panics when users update other package that use newer versions
+of [go.opentelemetry.io/otel/metric].
+
+Finally, an author can embed another implementation in theirs. The embedded
+implementation will be used for methods not defined by the author. For example,
+an author who want to default to silently dropping the call can use
```

Comment:
The only implementation that carries this guarantee is the NoOp because it is packaged with this API.  **_All_** other implementations can have the API get ahead of it, resulting in either fail to compile or panic.  

```suggestion
Finally, an author can embed the NoOp implementation. Any method not defined by
the author will default to take no action. For example, an author who wants to
default to silently dropping the call can use
```

Reply:
Sure, but the authors is still free to choose differently.

I can add a paragraph following this that explains why the no-op is the best choice here to help convey what you're saying. Sounds good?

Reply:
Updated, PTAL.

Reply:
No, if they choose any other implementation they are implicitly choosing one of these three. What ever they embed will have to make that choice for them.

If you can choose something else, but it's no different then choosing one of these options then we don't need to describe how it might work.

Reply:
> if they choose any other implementation they are implicitly choosing one of these three.

Sure, and both you and I can see that, but a new author may not infer this. Saying they have to embed the noop if they want to go this way will confuse the reader. They will rightfully think we missed a forth option and question the validity of the documentation.

Stating the explicit choice they can make here removes the confusion a new user can have otherwise.

Reply:
There is also the situation where the author wants to embed the OTel SDK. This is fine, and like you said it implies a choice for them to fail to compile, but explicitly they make the choice to embed something other than the noop.

Having clear documentation for new author is better than making the logical jump without them.

Reply:
Also, to be clear, there are really only two options if we want to talk about implicitly choosing behavior. Embedding the no-op means embedding an interface that will have compilation failures. It's just that we place it in a place were we can guarantee these don't occur for end users.